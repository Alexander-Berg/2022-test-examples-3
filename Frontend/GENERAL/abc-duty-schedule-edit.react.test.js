import React from 'react';
import { shallow } from 'enzyme';

import AbcDutyScheduleEdit from 'b:abc-duty-schedule-edit';

describe('AbcDutyScheduleEdit', () => {
    it('Should render schedule edit form', () => {
        const wrapper = shallow(
            <AbcDutyScheduleEdit
                people={{}}
                selectedPerson={null}
                absences={[]}
                schedules={{}}
                loading={false}
                roles={[]}
                currentRole={[null]}
                dateFrom={new Date(Date.UTC(2019, 1, 1))}
                dateTo={new Date(Date.UTC(2019, 1, 7))}
                onFilterChange={() => {}}
                onPersonChange={() => {}}
                isShiftEditOpen={false}
                shiftEditId={null}
                openShiftEdit={() => {}}
                closeShiftEdit={() => {}}
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });
});
