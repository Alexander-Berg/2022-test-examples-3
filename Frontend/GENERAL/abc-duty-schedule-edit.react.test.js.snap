// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbcDutyScheduleEdit Should render schedule edit form 1`] = `
<ModalLayout
  className="abc-duty-schedule-edit"
>
  <Wrapper>
    <Aside
      className="abc-duty-schedule-edit__aside"
      key="aside"
    >
      <abc-duty-schedule-edit__filter
        onChange={[Function]}
        roles={Array []}
        value={
          Array [
            null,
          ]
        }
      />
      <abc-duty-schedule-edit__people-list
        onChange={[Function]}
        people={Object {}}
        selected={null}
      />
    </Aside>
    <abc-duty-schedule-edit__main
      absences={Array []}
      dateFrom={2019-02-01T00:00:00.000Z}
      dateTo={2019-02-07T00:00:00.000Z}
      key="main"
      onOpenShiftEditClick={[Function]}
      schedules={Object {}}
    />
    <spin2
      key="spin"
      position="center"
      progress={false}
      size="m"
      tone="default"
      view="default"
    />
    <ShiftEditModal
      key="shiftEditModal"
      onCancel={[Function]}
      onCrossClick={[Function]}
      onSave={[Function]}
      scope="schedule-edit"
      shiftEditId={null}
    />
  </Wrapper>
</ModalLayout>
`;
